src/core/ext/xds/xds_http_filters.cc

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <grpc/support/port_platform.h>
+
+#include "src/core/ext/xds/xds_http_filters.h"
+
+#include "envoy/extensions/filters/http/router/v3/router.upbdefs.h"
+
+namespace grpc_core {
+
+const char* kXdsHttpRouterFilterConfigName =
+    "extensions.filters.http.router.v3.Router";
+
+namespace {
+
+class XdsHttpRouterFilter : public XdsHttpFilterImpl {
```

Comment:
Maybe this should be moved to something like `xds_http_route_filter.cc`. There will be more filters down the road, and it helps if we can limit the size of each file.

Reply:
In general, I do expect that individual filters will be implemented in their own files.  However, the router filter in particular is special, because it's basically required at the end of the filter chain, so I think it's fine to have it here.

Also, though, keep in mind that this class will not include the filter's actual functionality; that will be implemented in a C-core filter.  This class is just the glue necessary to allow a C-core filter to be configured via xDS.